src/main/java/com/microsoft/azure/servicebus/jms/AadAuthentication.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.servicebus.jms;
+
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+
+public class AadAuthentication {
+
+    private final String AUDIENCE = "https://servicebus.azure.net/.default";
+    private TokenCredential credential;
+    private AccessToken currentAccessToken;
+
+    /**
+     * Create an AadAuthentication using a token credential.
+     * @param credential. Can be of type Azure Default Credential, Msi credential or secret credential.
+     */
+	public AadAuthentication(TokenCredential credential) {
+		this.credential = credential;
+		this.currentAccessToken = this.generateAccessToken();
+	}
+	
+	/**
+     * Checks if current token is valid and generates a new token if is not valid 
+     * @return returns a valid token. 
+     */
+	public String getAadToken() {
+		//Generate new token if expire
+		if (this.currentAccessToken.isExpired()) {
+			synchronized(this.currentAccessToken) {
+				this.currentAccessToken = this.generateAccessToken(); 
+			}
+		}
```

Comment:
Why do you need this logic? Most of TokenCredential  implementation handle token cache and refreshing.
Here you can find TokenCredential implementations with suffix Credential- https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/src/main/java/com/azure/identity/ClientSecretCredential.java

Even if you want to use additional cache for some reason, I would prefer this instead https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/implementation/AccessTokenCache.java

Reply:
this will check the current token when reconnecting and will get a new token if is expired.